shared/js/dialer/keypad.js

```diff
@@ -460,42 +450,33 @@ var KeypadManager = {
       });
     }
 
+    this.restoreCaretPosition();
+
     // If it was a long press our work is already done
     if (this._longPress) {
       this._longPress = false;
       this._holdTimer = null;
-      this.restoreCaretPosition();
       return;
     }
 
     if (this._holdTimer) {
       clearTimeout(this._holdTimer);
     }
-
-    if (key === 'delete') {
-      this.restoreCaretPosition();
-    } else {
-      this.moveCaretToEnd(this.phoneNumberView);
-    }
```

Comment:
This effectively removes the last use of moveCaretToEnd(). If this function is not needed anymore you should remove it entirely.